File: chat-api/src/main/java/com/example/web_chat/PresentationLayer/DTO/Incoming/MessageRequestByTimestampDTO.java
package com.example.web_chat.PresentationLayer.DTO.Incoming;

public class MessageRequestByTimestampDTO
{

    private long creationTimestamp;

    private MessageRequestByTimestampType requestType;

    private int pageSize;
    
    private int pageNumber;

    public MessageRequestByTimestampDTO(long creationTimestamp, MessageRequestByTimestampType requestType, int pageSize, int pageNumber)
    {
        this.creationTimestamp = creationTimestamp;
        this.requestType = requestType;
        this.pageSize = pageSize;
        this.pageNumber = pageNumber;
    };

    public long getCreationTimestamp()
    {
        return this.creationTimestamp;
    }

    public MessageRequestByTimestampType getRequestType()
    {
        return this.requestType;
    }

    public int getPageSize()
    {
        return this.pageSize;
    }

    public int getPageNumber()
    {
        return this.pageNumber;
    }
}

File: chat-app/app/src/DTOs/MessageRequestByTimestampDTO.ts
import { MessageRequestByTimestampType } from "./MessageRequestByTimestampType";

export class MessageRequestByTimestampDTO
{
    private creationTimestamp: number;
    private requestType: MessageRequestByTimestampType;
    private pageSize: number;
    private pageNumber: number;

    constructor(
        creationTimestamp: number,
        requestType: MessageRequestByTimestampType,
        pageSize: number,
        pageNumber: number
    )
    {
        this.creationTimestamp = creationTimestamp;
        this.requestType = requestType;
        this.pageSize = pageSize;
        this.pageNumber = pageNumber;
    }

    getCreationTimestamp(): number
    {
        return this.creationTimestamp;
    }

    getRequestType(): MessageRequestByTimestampType
    {
        return this.requestType;
    }

    getPageSize(): number
    {
        return this.pageSize;
    }

    getPageNumber(): number
    {
        return this.pageNumber;
    }
}
